### DISCUSSION TITLE
What's the best way to add an external font?

### DISCUSSION THREAD

**Discussion by ttl256** [cotes2020/jekyll-theme-chirpy#1185] [Q&A] [answered]
I set the font I want in `_sass/variables-hook.scss` as described in the documentation.

```
$font-family-base: 'JetBrains Mono', monospace;
$font-family-heading: 'JetBrains Mono', monospace;
```

Now I need to somehow make a browser to download this font. I came up with copying the whole `_includes/head.html` and adding just this line and it works

```
<link href="https://fonts.googleapis.com/css?family=JetBrains+Mono" rel="stylesheet">
```

Is this the recommended way of introducing a custom font? I tried adding `_includes/cutsom-head.html` but it's ignored. I just don't like the idea of copying the whole file.

Thank you. (I use Starter if it matters)

**Comment by cotes2020** [comment, marked as answer]
There is no need to modify the `<head>`. You can define new fonts in `webfonts` inside one of the files in `_data/origin/` depending on how your site hosts its assets.

**Reply by ttl256** [reply]
Starter doesn't come with `_data/origin`. So I need to copy the directory from the main repo and edit the files. The main repo has two files: `basic.yml` and `cors.yml`.

As far as I can tell `basic.yml` refers to locally served fonts (static assets if I'm correct with the terminology), specifically to `/assets/lib/fonts/main.css` where it's possible to add/modify `@font-face`. And by adding one more `@font-face` I should be able to serve a font (of course I need to put the font itself alongside with `main.css`.

And `cors.yml` comes with `webfonts: https://fonts.googleapis.com/css2?family=Lato&family=Source+Sans+Pro:wght@400;600;700;900&display=swap` which makes a browser to download the font from Google. And adding `&family=JetBrains+Mono` to the value should do the job.

My conclusions come mostly from these lines in `head.html`:
```
{% if site.resources.ignore_env != jekyll.environment and site.resources.self_hosted %}
  <link href="{{ site.data.origin[type].webfonts | relative_url }}" rel="stylesheet">
```

Could you please validate my line of thoughts?
